File: EggIncognito.CodeGen/Templates/ruby/test_server.rb
require 'minitest/autorun'
require 'tmpdir'
require_relative 'server'

EID_PROTO = 'MhJFSTAwMDAwMDAwMDAwMDAwMDE='

class TestExtractEid < Minitest::Test
  def test_extract_eid
    assert_equal 'EI0000000000000001', extract_eid(EID_PROTO)
  end

  def test_extract_eid_empty
    assert_equal '', extract_eid('')
  end
end

class TestLoadFixture < Minitest::Test
  def setup
    @dir = Dir.mktmpdir('ei-test-')
    @orig = FIXTURES_PATH
  end

  def teardown
    FileUtils.rm_rf(@dir)
    silence_warnings { Object.const_set(:FIXTURES_PATH, @orig) }
  end

  def test_load_fixture_default
    FileUtils.mkdir_p(File.join(@dir, 'default'))
    File.binwrite(File.join(@dir, 'default', 'test.binpb'), "\x01\x02")
    silence_warnings { Object.const_set(:FIXTURES_PATH, @dir) }
    assert_equal "\x01\x02", load_fixture('test', '')
  end

  def test_load_fixture_eid_override
    FileUtils.mkdir_p(File.join(@dir, 'default'))
    FileUtils.mkdir_p(File.join(@dir, 'eids', 'EI0000000000000001'))
    File.binwrite(File.join(@dir, 'default', 'test.binpb'), "\x01")
    File.binwrite(File.join(@dir, 'eids', 'EI0000000000000001', 'test.binpb'), "\x02")
    silence_warnings { Object.const_set(:FIXTURES_PATH, @dir) }
    assert_equal "\x02", load_fixture('test', 'EI0000000000000001')
  end

  def test_load_fixture_missing
    silence_warnings { Object.const_set(:FIXTURES_PATH, @dir) }
    assert_equal '', load_fixture('nonexistent', '')
  end

  private

  def silence_warnings
    old = $VERBOSE
    $VERBOSE = nil
    yield
  ensure
    $VERBOSE = old
  end
end

File: EggIncognito.CodeGen/Templates/python/test_server.py
import os
import sys
import pytest

sys.path.insert(0, os.path.dirname(__file__))
import server

EID_PROTO = "MhJFSTAwMDAwMDAwMDAwMDAwMDE="


def test_extract_eid():
    assert server._extract_eid(EID_PROTO) == "EI0000000000000001"


def test_extract_eid_empty():
    assert server._extract_eid("") == ""


def test_load_fixture_default(tmp_path):
    (tmp_path / "default").mkdir()
    (tmp_path / "default" / "test.binpb").write_bytes(b"\x01\x02")

    orig = server.FIXTURES_PATH
    server.FIXTURES_PATH = str(tmp_path)
    try:
        data = server._load_fixture("test", "")
        assert data == b"\x01\x02"
    finally:
        server.FIXTURES_PATH = orig


def test_load_fixture_eid_override(tmp_path):
    (tmp_path / "default").mkdir()
    (tmp_path / "eids" / "EI0000000000000001").mkdir(parents=True)
    (tmp_path / "default" / "test.binpb").write_bytes(b"\x01")
    (tmp_path / "eids" / "EI0000000000000001" / "test.binpb").write_bytes(b"\x02")

    orig = server.FIXTURES_PATH
    server.FIXTURES_PATH = str(tmp_path)
    try:
        data = server._load_fixture("test", "EI0000000000000001")
        assert data == b"\x02"
    finally:
        server.FIXTURES_PATH = orig


def test_load_fixture_missing(tmp_path):
    orig = server.FIXTURES_PATH
    server.FIXTURES_PATH = str(tmp_path)
    try:
        data = server._load_fixture("nonexistent", "")
        assert data == b""
    finally:
        server.FIXTURES_PATH = orig
